integration-cli/docker_cli_build_test.go

```diff
@@ -4941,6 +4941,40 @@ func (s *DockerSuite) TestBuildLabelsCache(c *check.C) {
 
 }
 
+func (s *DockerSuite) TestBuildNotVerbose(c *check.C) {
+	// This test makes sure that the `-s` mode works correctly
+	testRequires(c, DaemonIsLinux)
+	tt := []struct {
+		TestName     string // Silent Build Name
+		BuildCmds    string // Build string
+		ShouldFail   bool   // Expect to fail
+		StdoutRegexp string
+		StderrRegexp string
+	}{
+		{"testsilentbuild_busybox_and_echo", "FROM busybox\nRUN echo hello", false, "^(sha256:|)[a-z0-9]{64}\\n$", "^$"},
+		{"testsilentbuild_invalid_image", "FROM blabla", true, "^$", "Step 1 : FROM blabla((\\n|\r)+.*)+Error: image library/blabla(:latest)? not found\\n"},
+		{"testsilentbuild_exit_code_1", "FROM busybox\nRUN echo hello\nRUN exit 1", true, "^$", "Step 1 : FROM busybox((\\n|\r)+.*)+RUN echo hello((\\n|\r)+.*)+hello((\\n|\r)+.*)+Step 3 : RUN exit 1((\\n|\r)+.*)+The command '/bin/sh -c exit 1' returned a non-zero code: 1"},
```

Comment:
I think it might make sense to split some of these points.

First, I don't think we should convert the json to text on the daemon because then the REST API output will look different when -q is used and there's an error.  I think a REST CLI will still expect JSON output.

In image.go just having:
```
if !buildConfig.Verbose && notVerboseBuffer.Len() > 0 {
    output.Write(notVerboseBuffer.Bytes())
}
```
Seems to work just fine for both REST CLIs and the Docker Cli.  I don't see the json output on the Docker CLI output like I did yesterday, but perhaps that's from another change you made, I'm not sure.

Second, the verbosity of the status messages. We might want to leave that for a new PR because the logic in jsonmessage already tries to take into account the non-tty case. While I would prefer that for each "id" we only showed the latest status instead of multiple, I can see why some people might like to see each state - e.g.:
```
6edcc89ed412: Verifying Checksum
6edcc89ed412: Download complete
6edcc89ed412: Pull complete
```
So, if we think that's too verbose then let's proposal that in a new PR.  But, in the end, if we can get:
```
docker build ... > out 2>&1
```
with and w/o `-q` to look the same when there's an error then I think we can claim success.

Finally, notice in the output you pasted if shows duplicate info:
```
6edcc89ed412: Pull complete
6edcc89ed412: Pull complete
```
and that's not due to the cli, its sent from the daemon.  I think this should be fixed, if possible, in a separate PR too.